graph/community/louvain_directed.go

```diff
@@ -353,25 +352,25 @@ func (g *ReducedDirected) To(v graph.Node) []graph.Node {
 func (g *ReducedDirected) HasEdgeBetween(x, y graph.Node) bool {
 	xid := x.ID()
 	yid := y.ID()
-	if xid == yid {
+	if xid == yid || xid != int64(int(xid)) || yid != int64(int(yid)) {
```

Comment:
I agree on an intuitive level that keeping the IDs in a machine word-sized value may have performance benefits for 32-bit platforms.

At the same time, without doing a benchmark on code using communities, we won't know for sure if keeping the node ID an `int` or changing it to `int64` actually has a huge impact. There may be other parts of the code which dominate the CPU profiles, such that the ID being in `int` or `int64` may not make a difference performance wise.

From my point of view, if the performance difference as measured by benchmarks and profiling is negligible, then focusing on readability and maintainability is more important.

Still, if there is an actually difference in measured performance when changing from `int` to `int64`, then I agree that it may be worth while to keep the internal community node IDs as `int`s.